core/domain/exp_domain.py

```diff
@@ -655,6 +659,75 @@ def validate(self):
         self.outcome.validate()
 
 
+class Hint(object):
+    """Value object representing a hint."""
+    def __init__(self, hint_text):
+        self.hint_text = html_cleaner.clean(hint_text)
+
+    def to_dict(self):
+        return {
+            'hint_text': self.hint_text,
+        }
+
+    @classmethod
+    def from_dict(cls, hint_dict):
+        return cls(html_cleaner.clean(hint_dict['hint_text']))
+
+    def validate(self):
+        if not isinstance(self.hint_text, basestring):
+            raise utils.ValidationError(
+                'Expected hint text to be a string, received %s' %
+                self.hint_text)
+
+
+class Solution(object):
+    """Value object representing a solution.
+
+    A solution consists of answer_is_exclusive, correct_answer and an
+    explanation.answer_is_exclusive when True indicates that it is the only
```

Comment:
Add space after "explanation." Also, reword:

"... explanation. When answer_is_exclusive is True, this indicates that ...; when it is False, this indicates that ..."